cmake/FindIgnOGRE.cmake

```
@@ -83,6 +83,16 @@ if (NOT WIN32)
 
   # loop through pkg config paths and find an ogre version that is < 2.0.0
   foreach(pkg_path ${PKG_CONFIG_PATH_TMP})
+
+    if (NOT EXISTS ${pkg_path})
+      continue()
+    endif()
+
+    unset(OGRE_FOUND)
+    unset(OGRE_INCLUDE_DIRS)
+    unset(OGRE_LIBRARY_DIRS)
+    unset(OGRE_LIBRARIES)
```

From the CMake [`unset` documentation](https://cmake.org/cmake/help/latest/command/unset.html):

> Note that when evaluating Variable References of the form `${VAR}`, CMake first searches for a normal variable with that name. If no such normal variable exists, CMake will then search for a cache entry with that name. Because of this unsetting a normal variable can expose a cache variable that was previously hidden. To force a variable reference of the form `${VAR}` to return an empty string, use `set(<variable> "")`, which clears the normal variable but leaves it defined.

I don't believe that the variables here should ever be cache variables, but for the sake of future-proofing the code and also for readability, should we change this to the following?
```suggestion
    set(OGRE_FOUND false)
    set(OGRE_INCLUDE_DIRS "")
    set(OGRE_LIBRARY_DIRS "")
    set(OGRE_LIBRARIES "")
```

---

yeah I think I noticed that during testing but I opted to follow [what is done](https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/cmake/FindIgnOGRE2.cmake#L278-L280) in FindIgnOGRE2:

I've now changed to setting the variable to empty strings instead of using `unset`, and verified that the variables are indeed empty strings each iteration so I'll stick with your suggestion. I didn't touch the `unset` code in `FindIgnOGRE2` though because don't have a windows machine to test that piece of code.

---

> I didn't touch the unset code in FindIgnOGRE2 though because don't have a windows machine to test that piece of code.

No problem :+1: maybe someone else with easy access to windows can test this. In the meantime, maybe we can at least add a note in `FindIgnOGRE2.cmake` that this could probably be changed to `set(...)` due to the intricacies of how variable references in the form of `${VAR}` work?

---

yep added your comment to `FindIgnOGRE2.cmake`. c8a5aea